README.md

```diff
@@ -20,95 +24,252 @@ var marked = transformer(require('jstransformer-marked'));
 
 var options = {};
 var res = marked.render('Some **markdown**', options);
-// => {body: 'Some <strong>markdown</strong>', dependencies: []}
+//=> {body: 'Some <strong>markdown</strong>', dependencies: []}
 ```
 
 This gives the same API regardless of the jstransformer passed in.
 
 ## API
+A transformer, once normalised using this module, will implement the following methods. Note that if the underlying transformer cannot be used to implement the functionality, it may ultimately just throw an error.
 
-A transformer, once normalised using this module, will implement the following methods.  Note that if the underlying transformer cannot be used to implement the functionality, it may ultimately just throw an error.
 
-### Returned object from `.render*`
+### Returned object from `.compile*`
+```js
+{fn: Function, dependencies: Array.<String>}
+```
+
+ - `fn` represents the render function, which also accept `locals`
+ - `dependencies` is an array of files that were read in as part of the compile process (or an empty array if there were no dependencies)
+
+
+### .compile
+Compile a given string and return an object.
+
+**Params**
+- `<str>` **{String}** string to compile
+- `[options]` **{Object}** optional options to pass to
+- `returns` **{Object}**
+
+**Requires the underlying transform to implement one of following methods**
+- `.compile`
+- `.render`
+
+**Example**
 
 ```js
-{body: String, dependencies: Array.<String>}
+transformer.compile(str, options);
+//=> {fn: Function, dependencies: Array.<String>}
 ```
 
- - `body` represents the result as a string
- - `dependencies` is an array of files that were read in as part of the render process (or an empty array if there were no dependencies)
 
-### `.render`
+### .compileAsync
+Compile a string asynchronously.
+
+**Params**
+- `<str>` **{String}** string to compile
+- `[options]` **{Object}** optional options to pass to
+- `[callback]` **{Function}** if not given, returns Promise
+- `returns` **{Promise}**
+
+**Requires the underlying transform to implement one of following methods**
+- `.compileAsync`
+- `.compile`
+- `.render`
+
+**Example**
 
 ```js
-transformer.render(str, options, locals);
-=> {body: String, dependencies: Array.<String>}
+transformer.compileAsync(str, options).then(function (res) {
+  //=> {fn: Function, dependencies: Array.<String>}
+});
 ```
 
-_requires the underlying transform to implement `.render` or `.compile`_
 
-Transform a string and return an object.
+### .compileFile
```

Comment:
Could use the following to make the headings look more like a function definition:
``` markdown
### `.compileFile`
```